include/live_model.h

```diff
@@ -220,6 +220,54 @@ namespace reinforcement_learning {
      */
     int report_outcome(const char* event_id, float outcome, api_status* status= nullptr);
 
+    /**
+     * @brief Report partial outcome of a decision based on an index.
+     *
+     * @param event_id  The unique event_id used when choosing an action should be presented here.  This is so that
+     *                  the action taken can be matched with feedback received.
+     * @param outcome Outcome as float.
+     * @param index Index of the partial outcome.
+     * @param status  Optional field with detailed string description if there is an error
+     * @return int Return error code.  This will also be returned in the api_status object
+     */
+    int report_outcome(const char* event_id, float outcome, int index, api_status* status= nullptr);
```

Comment:
[nit] do you think it is more intuitive to reorder to  event_id, index, outcome (event_id and index logically more like "Key" while outcome - "Value")

Reply:
Yeah, makes sense.